beacon_node/eth2-libp2p/src/peer_manager/mod.rs

```diff
@@ -263,6 +263,53 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
         }
     }
 
+    pub fn handle_rpc_error(&mut self, peer_id: &PeerId, protocol: Protocol, err: &RPCError) {
+        // map this error with a `PeerAction` asignated to the peer (if any), and decide if the
+        // peer manager should request some `PeerManagerEvent`
+        let (peer_action, event) = match err {
+            RPCError::IncompleteStream => {
+                // TODO: InvalidMessage? they Clogged? offline?
+                (None, None)
+            }
+            RPCError::InternalError(reason) => {
+                // NOTE: our fault. Now they hate us. Do nothing
+                (None, None)
+            }
+            RPCError::InvalidData => {
+                // TODO: invalid? malicious?
+                (None, None)
+            }
+            RPCError::IoError(e) => {
+                // TODO Whose fault? do nothing?
+                (None, None)
+            }
+            RPCError::RPCErrorResponse => {
+                // TODO what does this mean?
+                (None, None)
+            }
+            RPCError::SSZDecodeError(e) => {
+                // TODO their fault: InvalidMessage
+                (Some(PeerAction::InvalidMessage), None)
+            }
+            RPCError::UnsupportedProtocol => {
+                // TODO: check the protocol and decide
+                (None, None)
+            }
+            RPCError::StreamTimeout => {
+                // TODO: maybe they are dying?
```

Comment:
Yeah, this one could be they are slow, or our internet is slow. In this case i think we just lower their reputation a little.